Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp

```diff
@@ -0,0 +1,170 @@
+    if ( (step+1) % m_freq != 0 ) { return; }
+
+    // get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    // get MultiParticleContainer class object
+    auto & mypc = warpx.GetPartContainer();
+
+    // get number of species (int)
+    auto const nSpecies = mypc.nSpecies();
+
+    // get species names (std::vector<std::string>)
+    auto const species_names = mypc.GetSpeciesNames();
+
+    // get time at level 0
+    auto const t = warpx.gett_new(0);
+
+    // select species
+    int i_s;
+    for ( int i = 0; i < nSpecies; ++i )
+    {
+        if ( m_species_name == species_names[i] ) { i_s = i; }
+    }
+
+    // get WarpXParticleContainer class object
+    auto & myspc = mypc.GetParticleContainer(i_s);
+
+    using PType = typename WarpXParticleContainer::SuperParticleType;
+
+    // get parser
+    ParserWrapper<7> *fun_partparser = m_parser.get();
+
+    for ( int i = 0; i < m_bin_num; ++i )
+    {
+        // compute the histogram
+        m_data[i] = ReduceSum( myspc,
+        [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> Real
+        {
+            auto const w  = p.rdata(PIdx::w);
+            auto const x  = p.pos(0);
+            auto const y  = p.pos(1);
+            auto const z  = p.pos(2);
+            auto const ux = p.rdata(PIdx::ux);
+            auto const uy = p.rdata(PIdx::uy);
+            auto const uz = p.rdata(PIdx::uz);
+            auto const f = (*fun_partparser)(t,x,y,z,ux,uy,uz);
+            auto const f1 = m_bin_min + m_bin_size*i;
+            auto const f2 = m_bin_min + m_bin_size*(i+1);
+            if ( f > f1 && f < f2 )
+            {
+                if ( m_norm == "unity_particle_weight" )
+                { return 1.0; }
+                else
+                { return w; }
+            }
+            else
+            { return 0.0; }
+        });
+        // reduced sum over mpi ranks
+        ParallelDescriptor::ReduceRealSum
+            (m_data[i], ParallelDescriptor::IOProcessorNumber());
+    }
+
+    if ( m_norm == "max_to_unity" )
+    {
+        Real f_max = 0.0;
+        for ( int i = 0; i < m_bin_num; ++i )
+        {
+            if ( m_data[i] > f_max )
+            { f_max = m_data[i]; }
+        }
+        for ( int i = 0; i < m_bin_num; ++i )
+        {
+            if ( f_max > std::numeric_limits<Real>::min() )
+            { m_data[i] /= f_max; }
+        }
+    }
+    else if ( m_norm == "area_to_unity" )
+    {
+        Real f_area = 0.0;
+        for ( int i = 0; i < m_bin_num; ++i )
+        {
+            f_area += m_data[i] * m_bin_size;
+        }
+        for ( int i = 0; i < m_bin_num; ++i )
+        {
+            if ( f_area > std::numeric_limits<Real>::min() )
+            { m_data[i] /= f_area; }
+        }
+    }
+    else if ( m_norm == "default" )
+    { /* do nothing */ }
+    else if ( m_norm == "unity_particle_weight" )
+    { /* do nothing */ }
+    else { Abort("Unknown ParticleHistogram normalization type."); }
```

Comment:
Could you adopt a standard indentation here, e.g., replace
```c++
    if ( m_data[i] > f_max )
    { f_max = m_data[i]; }
 ```
with
```c++
    if ( m_data[i] > f_max )
        f_max = m_data[i];
```
or even
```c++
    if ( m_data[i] > f_max ) f_max = m_data[i];
```
or sth along those lines?